### ISSUE TITLE
A11y_AzureToolsStorageExplorer_SASGenerate_ScreenReader: NVDA not announces "Copy" control name in descriptive way.

### ISSUE THREAD

**Issue by Suraj1200499** [microsoft/AzureStorageExplorer#1913]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS1.3.1; #NVDA; #JAWS; #Win32; 

**Environment Details:**
Application Name: Azure Tools Storage Explorer
Application Version: 1.10.0
Windows10

**Repro Steps:**
1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and hit enter on "Create Blob container" to Create a new Blob container.
5. Once completed, Navigate to "Upload" button and Upload a blob/file by using  "Upload Files" control.
6. Blob gets uploaded, right click on blob and select "Get Shared Access Signature" control and hit enter.
7. Navigate to "Create" button and hit enter.
8. Start NVDA.
9. Navigate using Tab key to "Copy" control, verify NVDA not announces both "Copy" control names in descriptive way.

**​Actual:​**
While navigating using Tab key to "Copy" controls of disabled "URL", "Query string", NVDA not announces both "Copy" control names in descriptive way.

Same issue is observed in "**JAWS**" also.

**Expected:**
While navigating using Tab key to "Copy" controls of disabled "URL", "Query string", NVDA should announce both "Copy" control names in descriptive way.

**UserImpact:**
Screen reader users face difficulty to know which content needs to be copied if screen reader simply announces "Copy" for Copy button.

**Recommendation:**
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

**Attachment for Reference:**
[A11y_AzureToolsStorageExplorer_SASURL_ScreenReader.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3652406/A11y_AzureToolsStorageExplorer_SASURL_ScreenReader.pptx)

**Comment by MRayermannMSFT** [comment]
@Radha2019 @Charanrajkb @nirmalsahoo @Suraj1200499 @vinodbk197

Please verify whether or not this bug was fixed to your satisfaction in the latest release of Storage Explorer. You can obtain that release from https://www.storageexplorer.com

If the bug has not been fixed to your satisfaction, please:
- Explain how did the fix not meet your expectations.
- Tell us exactly how we can meet your expectations. If there is more than one way to do so, feel free to list the different ways.

**Comment by vinodbk197** [comment]
@MRayermannMSFT 
Verified this issue on:Azure Tools Storage Explorer (Version: 1.12.0)
The issue is partially resolved. For the second "copy" button, Screen reader announces incorrect description. Hence reopening this bug.

**Comment by craxal** [comment]
@vinodbk197 Can you be more specific, please? By "incorrect" description, do you mean that you're expecting NVDA to say "copy query string to clipboard" instead of "copy token to clipboard"?

**Comment by vinodbk197** [comment]
@craxal 
Yes, Nvda should announces as " Copy Query String to clipboard" instead of announcing "copy token to clipboard"

**Comment by Suraj1200499** [comment]
@Starmys RayermannMSFT Verified this issue in Private build of Windows.
Application version: 1.12.0.
This issue is no longer reproduced. Hence Closing the bug.
Added Closed Video in [One Drive](https://microsoft-my.sharepoint.com/personal/v-nisaho_microsoft_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fv%2Dnisaho%5Fmicrosoft%5Fcom%2FDocuments%2FStorageExplorer%5FReprovideos&originalPath=aHR0cHM6Ly9taWNyb3NvZnQtbXkuc2hhcmVwb2ludC5jb20vOmY6L3Avdi1uaXNhaG8vRXNTVUtnZm9PSzlIbUVMOFBxcVhkOTRCVXRBVkJ6S0pldHB5SDZwS0FzVXh5dz9ydGltZT1GRkRuQ25QRzEwZw) for reference.

**Comment by Suraj1200499** [comment]
@MRayermannMSFT Verified this issue in Private build of Windows.
Application version: 1.12.0.
This issue is no longer reproduced. Hence Closing the bug.
Added Closed Video in [One Drive](https://microsoft-my.sharepoint.com/personal/v-nisaho_microsoft_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fv%2Dnisaho%5Fmicrosoft%5Fcom%2FDocuments%2FStorageExplorer%5FReprovideos&originalPath=aHR0cHM6Ly9taWNyb3NvZnQtbXkuc2hhcmVwb2ludC5jb20vOmY6L3Avdi1uaXNhaG8vRXNTVUtnZm9PSzlIbUVMOFBxcVhkOTRCVXRBVkJ6S0pldHB5SDZwS0FzVXh5dz9ydGltZT1GRkRuQ25QRzEwZw) for reference.

**Comment by fsteffi** [comment]
GithubTags:#Closed;

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Win32-Sep2019;
